server/api/utils/singletons/k8s.py

```diff
@@ -615,11 +615,14 @@ def delete_secrets(
                 for key, value in k8s_secret.data.items()
                 if key not in secrets
             }
+        elif secrets is None:
+            # Delete all secrets
```

Comment:
Please also add to the docstring that when `secret=None`, all secrets are deleted